pkg/alertmanager/statefulset.go

```diff
@@ -425,7 +426,11 @@ func configSecretName(name string) string {
 }
 
 func volumeName(name string) string {
-	return fmt.Sprintf("%s-db", prefixedName(name))
+	v := fmt.Sprintf("%s-db", prefixedName(name))
+	if len(v) > maxNameLength {
+		return v[:maxNameLength]
```

Comment:
can kubernetes objects end with a `_` or `-`? This could definite produce something like this